modules/alloy/src/main/ts/ephox/alloy/ui/slider/SliderUi.ts

```diff
@@ -37,14 +37,18 @@ const sketch: CompositeSketchFactory<SliderDetail, SliderSpec> = (detail: Slider
   };
 
   const changeValue = (slider: AlloyComponent, newValue: SliderValue): Optional<boolean> => {
-    modelDetail.value.set(newValue);
-
+    setValue(slider, newValue);
     const thumb = getThumb(slider);
-    refresh(slider, thumb);
     detail.onChange(slider, thumb, newValue);
     return Optional.some<boolean>(true);
   };
 
+  const setValue = (slider: AlloyComponent, newValue: SliderValue): void => {
```

Comment:
This possibly should be above the `changeValue` function for readability, as `changeValue` uses `setValue`.